### ISSUE TITLE
pkgin struggling with nodejs's multiple version

### ISSUE THREAD

**Issue by sjorge** [NetBSDfr/pkgin#110]
I was trying to pin nodejs at a specific version using preferred.conf, and I discovered pkgin does not like the multiple versions.

Initially I thought the preferred.conf was massing things up, so I removed the nodejs entry... but no change.

```
[root@smtest ~]# pkgin search ^nodejs
nodejs-14.15.2       V8 JavaScript for clients and servers
nodejs-12.20.0       V8 JavaScript for clients and servers
nodejs-10.23.0       V8 JavaScript for clients and servers
nodejs-8.17.0nb4     V8 JavaScript for clients and servers

=: package is installed and up-to-date
<: package is installed but newer version is available
>: installed package has a greater version than available package
```

I wanted to pin on 12.*, but that aside... lets install.

```
[root@smtest ~]# pkgin -n in nodejs npm
No nodejs-8.17.0nb4 package available that satisfies preferred match nodejs>=14.*
calculating dependencies...done.

2 packages to install:
npm-6.14.9
nodejs-8.17.0nb4

0 to refresh, 0 to upgrade, 2 to install
8191K to download, 49M to install
```

It picks version 8... ouch thats the oldest! Not good.

```
[root@smtest ~]# pkgin pkg-descr nodejs
Information for https://pkgsrc.joyent.com/packages/SmartOS/trunk/x86_64/All/nodejs-14.15.2.tgz:
Description:
Node.js is an evented I/O framework for the V8 JavaScript engine. It is
intended for writing scalable network programs such as web servers.
This package holds the latest release.
Homepage:
https://nodejs.org/
```

Other pkgin commands do pick the latest ... ? I guess there is something wrong with the version matching.

```
[root@smtest ~]# pkgin in nodejs npm
No nodejs-8.17.0nb4 package available that satisfies preferred match nodejs>=14.*
calculating dependencies...done.

2 packages to install:
  npm-6.14.9 nodejs-8.17.0nb4

0 to refresh, 0 to upgrade, 2 to install
8191K to download, 49M to install

proceed ? [Y/n] y
nodejs-8.17.0nb4.tgz                                                                                 100% 8191KB   4.0MB/s   00:02
installing npm-6.14.9...
installing nodejs-8.17.0nb4...
pkg_install warnings: 0, errors: 0
reading local summary...
processing local summary...
marking nodejs-8.17.0nb4 as non auto-removable
marking npm-6.14.9 as non auto-removable
[root@smtest ~]# pkgin fug
nothing to do.
```

Having 8.x installed, doing a upgrade does not pick it up either.

```
[root@smtest ~]# pkgin in nodejs-14.15.2 npm
calculating dependencies...done.

1 package to upgrade:
  nodejs-14.15.2

0 to refresh, 1 to upgrade, 0 to install
0B to download, 20M to install

proceed ? [Y/n] y
upgrading nodejs-14.15.2...
pkg_install warnings: 0, errors: 0
reading local summary...
processing local summary...
```

The full version needs to be specified for it to install... it does not get downgrade afterwards though :)

**Comment by jperkin** [comment]
Should be fixed in https://github.com/NetBSDfr/pkgin/commit/4653f3cb484cc8f36d5b02fd4a459d54a4cd511a

**Comment by sjorge** [comment]
When will it hit trunk so I can verify?

**Comment by jperkin** [comment]
I want to add some tests first, but you should be able to build your own easily enough.

**Comment by jperkin** [comment]
FWIW, pkgin will never consider a package for upgrade if it doesn't have the same `PKGPATH`.  All of the nodejs versions have different `PKGPATH`s, so if you want to switch version after installing one previously, you will always have to specifically install the exact version you want.

**Comment by jperkin** [comment]
Pretty sure the original issue was fixed a few years ago, but 23.8.0 also has a number of selection fixes so I'm pretty confident this is now solved.
